daemon/create.go

```diff
@@ -78,6 +78,21 @@ func (daemon *Daemon) containerCreate(params types.ContainerCreateConfig, manage
 	return containertypes.ContainerCreateCreatedBody{ID: container.ID, Warnings: warnings}, nil
 }
 
+func setContainerIDEnv(container *container.Container, cidenv string) error {
+	if cidenv == "" {
+		return nil
+	}
+	// check if we already have an env var like that
+	for _, env := range container.Config.Env {
+		if strings.HasPrefix(env, cidenv) {
+			return errdefs.AlreadyExists(fmt.Errorf("Error environment variable %s already defined", cidenv))
```

Comment:
wrapped the returned error using `errdefs` to avoid the api complaining about unknown error type